plugins/arDominionB5Plugin/js/sidebarPaginatedList.js

```diff
@@ -0,0 +1,146 @@
+(function ($) {
+  "use strict";
+
+  $(() =>
+    $(".sidebar-paginated-list").each(
+      (_, element) => new paginatedListView($(element))
+    )
+  );
+
+  // Thresholds for the spinning timer and page input.
+  const BUSY_THRESHOLD = 200;
+  const PAGE_TYPING_THRESHOLD = 650;
+
+  class paginatedListView {
+    constructor($element) {
+      this.$element = $element;
+      this.url = this.$element.data("url");
+
+      this.currentPage = 1;
+      this.totalPages = parseInt(this.$element.data("total-pages"), 10);
+
+      // Stop execution when pagination is not needed
+      if (this.totalPages < 2) {
+        return;
+      }
+
+      var pageLinkBehavior = {
+        updatePageLinkState: function (disabled) {
+          this.parent().toggleClass("disabled", disabled);
+          this.attr("aria-disabled", disabled);
+        },
+      };
+      this.$prev = this.$element
+        .find(".page-link-prev")
+        .extend(pageLinkBehavior);
+      this.$next = this.$element
+        .find(".page-link-next")
+        .extend(pageLinkBehavior);
+
+      this.$prev.updatePageLinkState(true);
+
+      this.$pageInput = this.$element.find("input[type=number]");
+      this.$results = this.$element.find("> ul");
+      this.$spinner = this.$element.find(".spinner");
+      this.$resultStart = this.$element.find(".result-start");
+      this.$resultEnd = this.$element.find(".result-end");
+
+      this.init();
+    }
+    init() {
+      this.$next.on("click", this.next.bind(this));
+      this.$prev.on("click", this.prev.bind(this));
+      this.$pageInput.on("change", this.change.bind(this));
+    }
+    next(e) {
+      e.preventDefault();
+      this.fetchResults(this.currentPage + 1);
+    }
+    prev(e) {
+      e.preventDefault();
+      this.fetchResults(this.currentPage - 1);
+    }
+    // Setter/getter of busy state
+    busy(busy) {
+      // Getter
+      if (typeof busy === "undefined") {
+        return this._busy;
+      }
+
+      // Setter
+      this._busy = busy;
+
+      var $spinner = this.$spinner;
+
+      if (busy) {
+        this.$prev.updatePageLinkState(true);
+        this.$next.updatePageLinkState(true);
+
+        this.busyTimer && clearTimeout(this.busyTimer);
+        this.busyTimer = setTimeout(function () {
+          $spinner.removeClass("invisible");
```

Comment:
Where is this class coming from? Can we use `d-none` for consistency?

Reply:
I used invisible because I wanted the spinner to occupy the space (`visibility: hidden`).

Reply:
Now using d-none.